tests/php/Shortcodes/ImageShortcodeProviderTest.php

```diff
@@ -89,6 +89,24 @@ public function testShortcodeHandlerAddsDefaultAttributes()
             ))
         );
     }
+    
+    public function testShortcodeHandlerDoesNotResampleToNonIntegerImagesSizes()
+    {
+        $image = $this->objFromFixture(Image::class, 'imageWithoutTitle');
+        $parser = new ShortcodeParser();
+        $parser->register('image', [ImageShortcodeProvider::class, 'handle_shortcode']);
+
+        $this->assertEquals(
+            sprintf(
+                '<img src="%s" alt="" width="50%" height="auto">',
+                $image->Link()
+            ),
+            $parser->parse(sprintf(
+                '[image id="%d" alt="" width="50%" height="auto"]',
+                $image->ID
+            ))
+        );
```

Comment:
Test is failing because `sprintf` thinks the `%` in `50%` is meant to be another substitution. This should take care of that problem.

```suggestion
        $this->assertEquals(
            sprintf(
                '<img src="%s" alt="" width="50%%" height="auto">',
                $image->Link()
            ),
            $parser->parse(sprintf(
                '[image id="%d" alt="" width="50%%" height="auto"]',
                $image->ID
            ))
        );
```

Test is till failing because we end up with `alt="&quot;&quot;"` in the alt attribute, which must be related to that other issue you raise https://github.com/silverstripe/silverstripe-assets/issues/435

I'll see if I can figure out what's wrong with that one.

Reply:
Once the above suggestion is applied and https://github.com/silverstripe/silverstripe-framework/pull/9872 is merged, this PR should be green.